app/services/service_plans.rb

```diff
@@ -0,0 +1,24 @@
+class ServicePlans < TopologyServiceApi
+  def process
+    result = api_instance.list_service_offering_service_parameters_sets(service_offering_ref)
+    filter_result(result)
+  rescue StandardError => e
+    Rails.logger.error("Service Plans #{e.message}")
+    raise
+  end
+
+  def service_offering_ref
+    PortfolioItem.find_by!(:id => params['portfolio_item_id']).service_offering_ref
+  end
+
+  def filter_result(result)
```

Comment:
Possibly move this into a concern? Something like what @hsong-rh did [here](https://github.com/ManageIQ/approval-api/pull/2/files#diff-6dce1740add0245254336035060eb86e)